pandas/core/arrays/datetimes.py

```diff
@@ -1231,6 +1234,9 @@ def to_perioddelta(self, freq) -> TimedeltaArray:
         )
         from pandas.core.arrays.timedeltas import TimedeltaArray
 
+        if self._ndarray.dtype != "M8[ns]":
+            raise NotImplementedError("Only supported for nanosecond resolution.")
```

Comment:
any test hits this?

Reply:
no, but this is deprecated anyway.  mostly a note to myself not to bother implementing it for non-nano